packages/neon_framework/packages/account_repository/lib/src/account_repository.dart

```diff
@@ -0,0 +1,329 @@
+  Future<core.Status> getServerStatus(Uri serverURL) async {
+    final client = buildUnAuthenticatedClient(
+      httpClient: _httpClient,
+      userAgent: _userAgent,
+      serverURL: serverURL,
+    );
+
+    try {
+      final response = await client.authentication.core.getStatus();
+      return response.body;
+    } on http.ClientException catch (error, stackTrace) {
+      Error.throwWithStackTrace(FetchStatusFailure(error), stackTrace);
+    }
+  }
+
+  /// Initializes the Nextcloud login flow.
+  ///
+  /// May throw a [InitLoginFailure].
+  Future<Uri> loginFlowInit(Uri serverURL) async {
+    final client = buildUnAuthenticatedClient(
+      httpClient: _httpClient,
+      userAgent: _userAgent,
+      serverURL: serverURL,
+    );
+
+    try {
+      final initResponse = await client.authentication.clientFlowLoginV2.init();
+
+      final pollBody = core.ClientFlowLoginV2PollRequestApplicationJson((b) {
+        b.token = initResponse.body.poll.token;
+      });
+      _flow = (client, pollBody);
+
+      return Uri.parse(initResponse.body.login);
+    } on http.ClientException catch (error, stackTrace) {
+      Error.throwWithStackTrace(InitLoginFailure(error), stackTrace);
+    }
+  }
+
+  // TODO: update docs
+  /// Polls the login flow endpoint and returns the retrieved credentials.
+  ///
+  /// Throws a [StateError] if the login flow has not been initialized with [loginFlowInit].
+  /// Throws a [PollLoginFailure] when polling the endpoint fails. The flow must be initialized
+  /// again if a failure occurs.
+  Future<Credentials?> loginFlowPoll() async {
+    final flow = _flow;
+    if (flow == null) {
+      throw StateError('No login flow initialized');
+    }
+    final (client, body) = flow;
+
+    try {
+      final resultResponse = await client.authentication.clientFlowLoginV2.poll($body: body);
+
+      final response = resultResponse.body;
+      _flow = null;
+      return Credentials((b) {
+        b
+          ..serverURL = Uri.parse(response.server)
+          ..username = response.loginName
+          ..password = response.appPassword;
+      });
+    } on http.ClientException catch (error, stackTrace) {
+      if (error case DynamiteStatusCodeException(statusCode: 404)) {
+        return null;
+      }
+
+      _flow = null;
+      Error.throwWithStackTrace(PollLoginFailure(error), stackTrace);
+    }
+  }
+
+  /// Fetches the information for the account with the given [credentials].
+  ///
+  /// May throw a [FetchAccountFailure].
+  Future<Account> getAccount(Credentials credentials) async {
+    final client = buildClient(
+      credentials: credentials,
+      userAgent: _userAgent,
+      httpClient: _httpClient,
+    );
+
+    try {
+      final response = await client.authentication.users.getCurrentUser();
+
+      return Account((b) {
+        b
+          ..client = client
+          ..credentials.replace(credentials)
+          ..credentials.username = response.body.ocs.data.id;
+      });
```

Comment:
This is a bit messed up.
The Credentials already contain the username, but you still need to fetch it?
You probably meant for it to contain the loginName.
The loginName has to be used together with the appPassword because they are linked.
Using the username together with the appPassword should not work.

Reply:
Maybe Credentials should just have the loginName and Account should then have the additional username field

Reply:
This makes sense. Same as renaming the passowrd field to `appPassowrd`.
I'll change this in a followup as it would make a lot of changes in the neon code and requires thorough reviewing to make sure the display-name isn't used in a place, where the user id is expected.

This followup can also remove the `Account.password` helper method forcing consumers to access it through the credentials (`Account.credentials.password`).

Reply:
FYI: This is again behavior copied from the current code